[IMP] web: update documentation of patch function (#5017)
odoo/documentation: content/developer/reference/frontend/patching_code.rst

    },
  };

  patch(object, {
    fn() {
      // do things
    },
  });

When patching functions, we usually want to be able to access the ``parent``
function.  To do so, we can simply use the native ``super`` keyword:

.. code-block:: javascript

  patch(object, {
    fn() {
      super.fn(...arguments);
      // do other things
    },
  });

.. warning::

>     ``super`` can only be used in a method not a function. This means that syntaxes

Reviewer: `syntaxes` is a little weird. Maybe "the following constructs". Also maybe make this into a code block instead of inline code if possible, it will be easier to read, and you can add inline comments with the error that would be raised

Removing a patch
================

The `patch` function returns its counterpart. This is mostly useful for
testing purposes, when we patch something at the beginning of a test, and
unpatch it at the end.

.. code-block:: javascript

    const unpatch = patch(object, { ... });
    // test stuff here
    unpatch();

> Applying a same patch to multiple objects

Reviewer: "a same patch" => "the same patch"

=========================================

> It could happen that one wants to apply a same patch to multiple object but because

Reviewer: idem + "object" => "objects"

> of the way `super` keyword works, the `extension` has to be unique by patch and

Reviewer: ```suggestion
of the way the `super` keyword works, the `extension` can only be used for patching once and
```

cannot be copied/cloned (`check the documentation of the keyword <https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Operators/super#description>`_).
A function returning the object used to patch can be used to make it unique.

.. code-block:: javascript

    const obj1 = {};
    const obj2 = {};

    function createExtensionObj() {

>       return { ... };

Reviewer: add a dummy patch method using super

    }

    patch(obj1, createExtensionObj());
    patch(obj2, createExtensionObj());

.. warning::

  If an `extension` is based on another then the two extensions should
  be applied separately. Do not copy/clone an extension.

  .. code-block:: javascript

      const object = {};

>       const ext1 = { a: 1 };

Reviewer: make this example representative of the problem by actually using super

>       const invalid_ext2 = { ...ext1, b: 2 };

Reviewer: add a comment (eg `// this will not work: super will not refer to the correct object in methods coming from ext1`
